### ISSUE TITLE
Dice Score computing when gt is 0

### ISSUE THREAD

**Issue by JiYuanFeng** [Project-MONAI/MONAI#4052]
Hi, thanks for your contributions, when using the DiceMetric calculation, I found that when the gt of a class is 0, and the prediction is not 0, the returned dice value is NAN, and from the original [Dice calculation formulation](https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient), the returned value should be 0 because the denominator is not 0.

from monai.metrics import DiceMetric  
gt = torch.zeros((1, 3, 512, 512))  
pred = torch.zeros((1, 3, 512, 512))  
pred[:, 2, :, :] += 1  
dice_metric = DiceMetric(include_background=True, reduction="mean_batch", get_not_nans=True)  
for r, g in zip(pred, gt):  
  dice_metric(y_pred=r.unsqueeze(0), y=g.unsqueeze(0))  
dice, non_nan_count = dice_metric.aggregate()

Now, the output is `tensor([0., 0., 0.]) and tensor([0., 0., 0.])`

In the case of gt is empty, the error caused by false positives seems to be ignored

from monai.metrics import DiceMetric
gt = torch.zeros((1, 3, 512, 512))
gt[0, 2, :, :] += 1
pred = torch.zeros((1, 3, 512, 512))
pred[:, 2, :, :] += 1
dice_metric = DiceMetric(include_background=True, reduction="mean_batch", get_not_nans=True)
for r, g in zip(pred, gt):
    dice_metric(y_pred=r.unsqueeze(0), y=g.unsqueeze(0))
dice, non_nan_count = dice_metric.aggregate() 

the output is `tensor([0., 0., 1.]) and tensor([0., 0., 1.])`

**Comment by Nic-Ma** [comment]
Hi @dongyang0122 ,

Could you please take a look and share your research experience on it?

Thanks in advance.

**Comment by dongyang0122** [comment]
@JiYuanFeng it is not straightforward to compute Dice score on the class without any foreground pixels/voxels. The ideal prediction (no foreground prediction) should be with Dice score 1 via zero divided by zero. It may introduce numerical instability during the computation. So we choose to skip that class when the gt foreground is empty (until seeing more convincing discussion on this matter). Moreover, Dice score is not designed to detect false positives specifically. It would be better to have another metric function measuring the false positives/negatives.

**Comment by JiYuanFeng** [comment]
Hi, I also survey the dice metric implementation of some medical segmentation benchmarks. 
Here is the metric used in MSD challenge:

<img width="1111" alt="image" src="https://user-images.githubusercontent.com/20653249/161694989-24ff2e1d-0c23-41ab-a1b4-61cd53325c91.png">

They only ignore the cases where both gt and pred are both empty.

**Comment by dongyang0122** [comment]
@JiYuanFeng thanks for your explanation! I can understand the reason to skip cases where both gt and preds are zero. However, if we compute the Dice score when gt is all zero, then the Dice score become either zero or NaN. Its value is no longer a scalar. It is possible that it does not tell us how good the model is on this specific case, even when model is somehow improved on other cases via fine-tuning.
Penalties for incorrect predictions of all-zero gt need to be computed. Often it can be reflected in other classes of dice scores (e.g., background).

**Comment by JiYuanFeng** [comment]
Thanks for your explanation again!
